### ISSUE TITLE
Signalétique - Champ `pictogramme` des lignes à transformer en liste de valeurs

### ISSUE THREAD

**Issue by IdrissaD** [GeotrekCE/Geotrek-admin#3327]
Actuellement le champ `pictogramme` est un champ textuel sans contrainte. Par définition, un pictogramme est un symbole qui est réutilisé à de nombreuses reprises sur des panneaux. Le renseignement libre de ce champ sous forme de texte comporte le risque d'erreurs de saisie et de non homogénéité du nom des pictogrammes.

Un modèle `SignagePictogram` avec nom, image, code et description pour chaque pictogramme permettrait de faciliter la saisie des lignes et de limiter les erreurs.
Plusieurs pictogrammes devraient pouvoir être sélectionnés, car une ligne peut en contenir plusieurs.

**Comment by LePetitTim** [comment]
@camillemonchicourt 

Le dev a été commencé, il me reste a savoir quoi afficher dans la fiche détail des signalétiques et lames ainsi que dans les fichiers d'exports.

**Comment by camillemonchicourt** [comment]
Ouais et je m'interroge comment on peut gérer les éventuelles données existantes pour ceux qui ont déjà saisis des lignes ?

Il faudrait aussi peut-être des fixtures pour avoir une première base de pictos par défaut ?

Chaque picto ayant un nom, dans les exports, on peut afficher ce nom. Ah moins qu'ils aient besoin du picto dans les exports pour les commandes ?
Sur les fiches détails, c'est bien d'afficher le visuel et le nom, le code, la description au survol ?

**Comment by IdrissaD** [comment]
Effectivement certaines organisations telles que, de mémoire, le CD du Gard ont beaucoup utilisé ce champ, ce qui posera sûrement problème lors de la mise à jour de GA.

Pour les fixtures, il y aurait la liste des pictogrammes officiels des Parcs nationaux (image, label et code), il y a également les pictos de la FFRP et ceux de la FFC (notamment pour le VTT). Peut-être pas besoin d'intégrer tous les pictos des PN car il y en a plusieurs dizaines dont beaucoup qui sont rarement utilisés, une sélection peut être faite (cf. https://www.ecrins-parcnational.fr/sites/ecrins-parcnational.com/files/fiche_doc/12785/chartesignapedestrepnfpne2014.pdf page 71).
Un mix de ces trois sources de pictogrammes devrait couvrir la grande majorité des usages à mon avis.

D'accord avec Camille pour les fiches détails, le picto en lui-même est sûrement le plus important à afficher. Les noms peuvent être assez longs (ex: "Pas de pêche à moins de cinquante mètres du rivage"), et les codes assez obscurs (ex: "023"), mais s'ils pouvaient apparaître au survol du picto ce serait idéal.

**Comment by AudreyRemy** [comment]
Bonjour,
Nous avons utilisé le champ picto pour toute notre signalétique. Je vous rejoins complètement sur le risque d'erreur. 
Sera-t-il possible de personnaliser la liste des pictos via l'admin du Geotrek-admin ?

**Comment by babastienne** [comment]
Nous sommes en train de travailler à l'ajout de cette fonctionnalité. Toutefois nous aurions besoin pour remplir les fixtures des fichiers sources et de la pré-sélection évoquée par @IdrissaD .

Serait-il possible de nous transmettre une liste de pictogramme qui selon vous couvre l'essentiel des besoins des territoires ? Nous n'avons pas pu trouver de sources pour les pictogrammes des PN évoqués dans la charte graphique des signalétiques.

**Comment by IdrissaD** [comment]
@amandine-sahl et @Mila-Taillandier sont en train de travailler sur une proposition de fixtures pour les règles des zones de sensibilité (règles qui comprennent des pictos). Les deux fixtures ayant de nombreux éléments communs (même si moins de champs pour les pictos), je les laisse prendre le relais là-dessus

**Comment by IdrissaD** [comment]
Voici une sélection de pictogrammes au format SVG, en ce qui concerne la réglementation des Parcs nationaux :
[PICT_REGL_PNX_SVG.zip](https://github.com/GeotrekCE/Geotrek-admin/files/12251554/PICT_REGL_PNX_SVG.zip)

**Comment by camillemonchicourt** [comment]
En effet on va fournir dès que possible une fixture quasi prête, pour la table des règles ajoutée récemment (https://github.com/GeotrekCE/Geotrek-admin/issues/3470).
Elle se base sur les mêmes pictos des règles des parcs nationaux.
A voir si il faut utiliser les mêmes pictos (sans les dupliquer ?) pour le champs des pictogrammes des lignes de signalétique ?
Et à voir si c'est pas trop spécifique à la signalétique des parcs nationaux ?

A voir aussi si cette fixture peut être isolée pour pouvoir être déployée, même sur un Geotrek-admin déjà existant, en lançant une commande après mise à jour ?

**Comment by babastienne** [comment]
Ok merci pour vos retours et merci d'avance pour compléter les informations.

Pour le moment j'ai l'impression que les pictos signalétique proposés sont très spécifiques aux Parcs Nationaux, ca serait bien d'en avoir aussi des plus génériques je pense.

**Comment by AudreyRemy** [comment]
En plus des pictos fédéraux comme GR, PR, GRP, VTT, GTVTT ... nous utilisons le picto jacquaire (la coquille) et des pictos "activité" devant le temps de marche (Marche, vtt, équestre, ...). Nous n'avons pas encore ces pictos car nous avons lancé un appel d'offre sur la charte graphique de la signalétique.
Merci pour tout ce travail.

**Comment by IdrissaD** [comment]
> Ok merci pour vos retours et merci d'avance pour compléter les informations.
> 
> Pour le moment j'ai l'impression que les pictos signalétique proposés sont très spécifiques aux Parcs Nationaux, ca serait bien d'en avoir aussi des plus génériques je pense.

Oui j'avais mentionné les pictos de la FFC et de la FFRP qui sont importants à ajouter, mais je n'y ai pas accès donc je n'ai pu fournir que ceux des PN.

**Comment by babastienne** [comment]
La release [2.100.0](https://github.com/GeotrekCE/Geotrek-admin/releases/tag/2.100.0) embarque la transformation du champ pictogramme sous forme de liste de valeurs configurables.

Il reste à ajouter une ou des fixtures pour proposer aux utilisateurs des valeurs par défaut.

**Comment by IdrissaD** [comment]
Merci pour le développement ! Plusieurs remarques suite à des essais sur l'instance de démo des Écrins : 

- il faudrait clarifier l'affichage des champs des pictos dans les fiches détails, car pour l'instant le nom, le code et la description sont affichées, ce qui complique la lisibilité (cf. image ci-dessous). Je propose de n'afficher que le nom du picto, et éventuelle le code, la description n'ayant pas d'intérêt à apparaître ici.
![image](https://github.com/GeotrekCE/Geotrek-admin/assets/73114156/93d941d9-9a5e-4939-87f5-d27f3fb22a71)

- idéalement il faudrait afficher les images des pictos dans la fiche détail, par exemple juste avant le nom, cela rendrait à mon avis l'interface plus lisible

- la taille par défaut des images des pictogrammes dans l'interface d'admin Django est trop grande, et ce qu'elle que soit la taille réelle des images téléversées : 
![image](https://github.com/GeotrekCE/Geotrek-admin/assets/73114156/86e0fbdf-db79-4da5-9fe2-9395897f3f93)

**Comment by babastienne** [comment]
@IdrissaD je suis d'accord avec toi pour afficher moins d'infos mais afficher en revanche le picto dans la fiche détail. Ne faudrait-il pas également afficher le picto dans la liste déroulante lorsque l'utilisateur sélectionne une valeur au moment de la saisie ?

Par contre je ne reproduit pas ton problème sur la taille des images. J'ai essayé avec deux pictos de tailles vraiment différentes et les images sont bien importées avec leur taille sans redimensionnement.

<details><summary>Les images utilisées pour mes tests</summary>
<p>

Image 1 : 

![canoe V2 32](https://github.com/GeotrekCE/Geotrek-admin/assets/16348048/93e7fcbb-d540-4587-96e9-c74b7c26b62e)

Image 2 : 

![RANDO SUR L'EAUPICTOS](https://github.com/GeotrekCE/Geotrek-admin/assets/16348048/39e9c418-703a-48ff-8a52-e7987697dcc9)


</p>
</details>

**Comment by IdrissaD** [comment]
Effectivement, c'était parce que j'avais téléversé les images au format SVG !

Je n'y avais pas pensé pour la liste déroulante, à voir si ça n'augmente pas trop la hauteur des lignes, comme c'est une liste qui risque déjà de comporter beaucoup de valeurs.
